### PR TITLE
Pretty checkbox, data sharing check on feedback.

### PR DESCRIPTION
**Pull request by mjvkeulen**
## Description
- Added a pretty checkbox component, this should allow for a high level of customization. Currently the default is set to our dark-blue theme colour. However, images/svg or animations are possible. For a full list of possibilities see: https://hamed-ehtesham.github.io/pretty-checkbox-vue/
- Added a disclaimer to our feedback functionality, asking the user for permission to share their username, first name, email and feedback with our email provider.

image
image

vs:

image
image

Currently using this:

image

image

### CODE REVIEW

File: src/vue/src/components/assets/Feedback.vue [hunk 1]
```diff
@@ -88,6 +95,8 @@ export default {
                 this.$toasted.error('Please choose a feedback type')
             } else if (this.feedback === '') {
                 this.$toasted.error('Please describe your feedback')
+            } else if (!this.privacyAgreement) {
+                this.$toasted.error('You will have to agree to share, in order for us to read your feedback!')
```
Comment (Larspolo) [inline src/vue/src/components/assets/Feedback.vue]: agree to the privacy agreement

File: src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@import '~sass/vendor/pretty-checkbox/scss/states/indeterminate';
+@import '~sass/vendor/pretty-checkbox/scss/extras/toggle';
+@import '~sass/vendor/pretty-checkbox/scss/extras/plain';
+@import '~sass/vendor/pretty-checkbox/scss/extras/round';
+@import '~sass/vendor/pretty-checkbox/scss/extras/curve';
+@import '~sass/vendor/pretty-checkbox/scss/extras/animation';
+@import '~sass/vendor/pretty-checkbox/scss/extras/disabled';
+@import '~sass/vendor/pretty-checkbox/scss/extras/locked';
+@import '~sass/vendor/pretty-checkbox/scss/extras/colors';
+@import '~sass/vendor/pretty-checkbox/scss/extras/bigger';
```
Comment (Larspolo) [inline src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss]: Maybe there is an npm module for this?

Can these files not just be generated in the makefile iso included in the git? This pollutes out repo a bit too much imo

Comment (mjvkeulen) [reply src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss]: Styling the checkbox required adapting the scss directly. Therefore it is different compared to the default settings of a npm install. If anyone else wants different settings or to make use of this versatile component, direct control is again required.

Comment (engelhamer) [reply src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss]: Kinda bothers me that the sass folder now contains scss though.

File: src/vue/src/components/assets/Feedback.vue [hunk 3]
```diff
@@ -36,16 +36,21 @@
             @change="filesHandler"
             placeholder="Choose a file...">
         </b-form-file>
+        <pretty-checkbox v-model="privacyAgreement" class="p-default p-fill multi-form">
+            I agree to share my username, email, first name and feedback with eJournal's email provider zoho.
```
Comment (engelhamer) [inline src/vue/src/components/assets/Feedback.vue]: Please capitalize the company name, as well as maybe add a hyperlink to it?

File: src/vue/src/components/assets/Feedback.vue [hunk 4]
```diff
@@ -36,16 +36,21 @@
             @change="filesHandler"
             placeholder="Choose a file...">
         </b-form-file>
+        <pretty-checkbox v-model="privacyAgreement" class="p-default p-fill multi-form">
```
Comment (engelhamer) [inline src/vue/src/components/assets/Feedback.vue]: Please add a mandatory sign to this.

Comment (engelhamer) [reply src/vue/src/components/assets/Feedback.vue]: `*`

Comment (engelhamer) [reply src/vue/src/components/assets/Feedback.vue]: Maybe to the other fields as well.

**Comment by engelhamer** [comment]
Would like to see a screenshot as well, though I suspect we will make some alterations in the future anyways.

File: src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss (hunk 2 above)
Comment (Larspolo) [reply src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss]: Maybe you can add an extra class with !important iso altering the source code?

Comment (mjvkeulen) [reply src/vue/src/sass/vendor/pretty-checkbox/pretty-checkbox.scss]: I agree with the name of sass. However its simply our stylesheet folder, so it is not out of place. I could rename it.

File: src/vue/src/components/assets/Feedback.vue (hunk 3 above)
Comment (mjvkeulen) [reply src/vue/src/components/assets/Feedback.vue]: The entire text is part of the input, so a hyperlink won't work. I will capitalize zoho however.

File: src/vue/src/components/assets/Feedback.vue [hunk 5]
```diff
@@ -121,6 +137,13 @@ export default {
...
         }
     }
-
 }
 </script>
+
+<style lang="scss">
+// This could be prevented by converting the scss to sass, then importing our colors to the pretty-checkbox settings
+// followed by overwriting the defaults. (The variables have to be set before the import for this to work, same
+// language scope.)
```
Comment (Larspolo) [inline src/vue/src/components/assets/Feedback.vue]: Both ways are not that nice, I like this more

**Comment by engelhamer** [comment]
Looks good!

**Pull request merged**